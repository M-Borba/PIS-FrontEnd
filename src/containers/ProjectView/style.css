.react-calendar-timeline {
  margin: 1%;
  Box-shadow: -10px 10px 30px 2px rgba(0, 0, 0, 0.08);
}

.react-calendar-timeline .rct-dateHeader-primary {
  background-color: initial;
  border-left: 1px solid #bbb;
  border-right: 1px solid #bbb;
  color: #fff;
  font-size: 16px;
  font-weight: 700 !important;
}

.rct-item {
  line-height: 20px !important;
  height: 20px !important;
  margin-top: 5px !important;
  border-radius: 10px !important;
  border: 0 !important;
}

.rct-item-content {
  padding: 0 10px !important;
}

.react-calendar-timeline .rct-dateHeader {
  font-size: 16px;
  font-weight: 400;
  border: 0;
}

.react-calendar-timeline .rct-header-root {
  background: #242424;
  border-bottom: 1px solid #F1EDED;
}

.react-calendar-timeline .rct-vertical-lines .rct-vl {
    border: 0;
}

.react-calendar-timeline .rct-sidebar {
    border-right: 1px solid #F1EDED;
}

.react-calendar-timeline .rct-sidebar .rct-sidebar-row {
    border-bottom: 1px solid #F1EDED;
}

.react-calendar-timeline .rct-horizontal-lines .rct-hl-even, .react-calendar-timeline .rct-horizontal-lines .rct-hl-odd {
    border-bottom: 1px solid #F1EDED;
    background: #FAFAFA;
}

.react-calendar-timeline .rct-sidebar .rct-sidebar-row.rct-sidebar-row-odd,
.react-calendar-timeline .rct-sidebar .rct-sidebar-row.rct-sidebar-row-even {
    background: #FAFAFA;
    padding-left: 23px;
    min-height: 40px;
}

.react-calendar-timeline .rct-sidebar {
    box-shadow: -10px 10px 30px 2px rgba(0, 0, 0, 0.05);
    z-index: 10;
}

.react-calendar-timeline .rct-calendar-header {
    box-shadow: 0px 10px 30px 2px rgb(0 0 0 / 5%);
    z-index: 1;
}
