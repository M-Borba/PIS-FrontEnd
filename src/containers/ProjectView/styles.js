import { makeStyles } from "@material-ui/core/styles";

export const useStyles = makeStyles((theme) => ({
  modal: {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    background: "white",
    border: "2px solid #000",
    boxShadow: 24,
    maxHeight: "85vh",
    overflow: "auto",
    p: 4,
  },
  modalInfo: {
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
    width: "70%",
    maxWidth: "800px",
    background: "white",
    boxShadow: 24,
    overflowX: "hidden",
  },
  closeButton: {
    position: "absolute",
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
  box: {
    display: "flex",
    padding: 8,
  },
}));
