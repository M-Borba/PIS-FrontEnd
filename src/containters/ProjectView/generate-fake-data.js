import randomColor from "randomcolor";
import moment from "moment";

export default function (groupCount = 30, itemCount = 30) {
  let randomSeed = Math.floor(Math.random() * 1000);
  let groups = [];
  for (let i = 0; i < groupCount; i++) {
    groups.push({
      id: `${i + 1}`,
      title: "Proyecto " + `${i + 1}`,
      bgColor: randomColor({ luminosity: "light", seed: randomSeed + i }),
    });
  }
  let items = [];
  for (let i = 0; i < itemCount; i++) {
    const startDate = new Date(2021, 8, 1, 0, 0); //dia de setiembre
    const day = 86400000 * Math.floor(Math.random() * 30);
    const startValue = moment(startDate).valueOf() + day; //dia random del mes de setiembre
    const endValue = moment(startValue + 86400000 * 20).valueOf(); //86400000 = 24h (proyecto duracion 20 dias)

    items.push({
      id: i + "",
      group: i + "",
      start: startValue,
      end: endValue,
      canMove: startValue > new Date().getTime(),
      canResize: "both",
      className:
        moment(startDate).day() === 6 || moment(startDate).day() === 0
          ? "item-weekend"
          : "",
    });
  }

  items = items.sort((a, b) => b - a);

  return { groups, items };
}
