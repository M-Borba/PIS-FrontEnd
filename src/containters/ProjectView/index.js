import React, { Component } from "react";
import moment from "moment";

import Timeline, {
  TimelineHeaders,
  SidebarHeader,
  DateHeader,
} from "react-calendar-timeline";

import "react-calendar-timeline/lib/Timeline.css";

import Switcher from "../../components/Switcher/index";

import PropTypes from "prop-types";

import generateFakeData from "./generate-fake-data";

var keys = {
  groupIdKey: "id",
  groupTitleKey: "title",
  groupRightTitleKey: "rightTitle",
  itemIdKey: "id",
  itemTitleKey: "title",
  itemDivTitleKey: "title",
  itemGroupKey: "group",
  itemTimeStartKey: "start",
  itemTimeEndKey: "end",
  groupLabelKey: "title",
};

const customTimeSteps = {
  second: 0,
  minute: 0,
  hour: 0,
  day: 1,
  month: 1,
  year: 1,
};


export default class ProjectTimeline extends Component {
  constructor(props) {
    super(props);

    const { groups, items } = generateFakeData();
    const defaultTimeStart = moment().startOf("day").toDate();
    const defaultTimeEnd = moment().startOf("day").add(1, "day").toDate();

    this.state = {
      groups,
      items,
      defaultTimeStart,
      defaultTimeEnd,
    };
  }

  handleItemMove(itemId, dragTime, newGroupOrder) {
    const { items, groups } = this.state;

    const group = groups[newGroupOrder];

    this.setState({
      items: items.map((item) =>
        item.id === itemId
          ? Object.assign({}, item, {
            start: dragTime,
            end: dragTime + (item.end - item.start),
            group: group.id,
          })
          : item
      ),
    });

    console.log("Moved", itemId, dragTime, newGroupOrder);
  };

  handleItemResize(itemId, time, edge) {
    const { items } = this.state;

    this.setState({
      items: items.map((item) =>
        item.id === itemId
          ? Object.assign({}, item, {
            start: edge === "left" ? time : item.start,
            end: edge === "left" ? item.end : time,
          })
          : item
      ),
    });

    console.log("Resized", itemId, time, edge);
  };

  render() {
    const { groups, items, defaultTimeStart, defaultTimeEnd } = this.state;

    return (
      <Timeline
        groups={groups}
        items={items}
        keys={keys}
        fullUpdate
        itemTouchSendsClick={true}
        dragSnap={60 * 60 * 24 * 1000} //dia
        timeSteps={customTimeSteps}
        itemHeightRatio={0.75}
        canMove={true} //se pueden mover
        canChangeGroup={false} //no se pueden "cambiar de renglon"
        canResize={"both"}
        defaultTimeStart={defaultTimeStart}
        defaultTimeEnd={defaultTimeEnd}
        onItemMove={this.handleItemMove}
        onItemResize={this.handleItemResize}
        sidebarContent={<div>pe√°r</div>}
        sidebarWidth={200}
      >
        <TimelineHeaders className="sticky">
          <SidebarHeader>
            {({ getRootProps }) => {
              return (
                <div {...getRootProps()}>
                  <Switcher
                    onSwitch={this.props.onSwitch}
                    isProjectView={this.state.isProjectView}
                  />
                </div>
              );
            }}
          </SidebarHeader>
          <DateHeader unit="primaryHeader" />
          <DateHeader />
        </TimelineHeaders>
      </Timeline>
    );
  }
}

ProjectTimeline.propTypes = {
  isProjectView: PropTypes.bool,
  onSwitch: PropTypes.func,
};
