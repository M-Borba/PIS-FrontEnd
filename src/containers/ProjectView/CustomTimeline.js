import React, { useState, useEffect, Fragment } from "react";
import moment from "moment";
import Modal from "@material-ui/core/Modal";
import { IconButton, Box } from "@material-ui/core";
import CloseIcon from "@material-ui/icons/Close";
import Timeline, {
  TimelineHeaders,
  SidebarHeader,
  DateHeader,
} from "react-calendar-timeline";
import randomColor from "randomcolor";
import { axiosInstance } from "../../config/axios";
import Switcher from "../../components/Switcher/";
import PropTypes from "prop-types";
import "./style.css";
import { useStyles } from "./styles";

import InfoProyecto from "../../containers/InfoProyecto";
import FilterForm from "../../components/FilterForm";
import Notificacion from "../../components/Notificacion";

var keys = {
  groupIdKey: "id",
  groupTitleKey: "title",
  groupRightTitleKey: "rightTitle",
  itemIdKey: "id",
  itemTitleKey: "title",
  itemDivTitleKey: "title",
  itemGroupKey: "group",
  itemTimeStartKey: "start",
  itemTimeEndKey: "end",
  groupLabelKey: "title",
};

ProjectTimeline.propTypes = {
  onSwitch: PropTypes.func,
  isProjectView: PropTypes.bool,
};

export default function ProjectTimeline({ onSwitch, isProjectView }) {
  const [groups, setGroups] = useState([]);
  const [items, setItems] = useState([]);
  const [projectData, setProjectData] = useState([]);
  const [openInfo, setOpenInfo] = React.useState(false);
  const [filteredData, setFilteredData] = useState([true]);
  const [fetchingError, setFetchingError] = useState([false]);
  const [filters, setFilters] = useState({
    project_type: "",
    project_state: "",
    organization: "",
  });

  const classes = useStyles();

  const handleInfoClose = () => {
    setOpenInfo(false);
  };

  var groupsToAdd = [];
  var itemsToAdd = [];
  const [notify, setNotify] = useState({
    isOpen: false,
    message: "",
    type: "success",
    reload: false,
  });
  const customTimeSteps = {
    second: 0,
    minute: 0,
    hour: 0,
    day: 1,
    month: 1,
    year: 1,
  };

  const fetchData = (filterParams = {}) => {
    // to avoid sending empty query params
    for (let key in filterParams) {
      if (filterParams[key] === "" || filterParams[key] === null) {
        delete filterParams[key];
      }
    }
    axiosInstance
      .get("/projects", { params: filterParams })
      .then((response) => {
        const rows = response.data.projects;
        if (rows.length == 0) {
          setFilteredData(false);
          setNotify({
            ...notify,
            isOpen: true,
            message: "No existen datos para los filtros seleccionados",
            type: "error",
          });
        }
        rows.map((proj) => {
          setFilteredData(true);
          setFetchingError(false);

          groupsToAdd.push({
            id: proj.id,
            title: proj.name,
            bgColor: randomColor({ luminosity: "light" }),
          });
          setGroups(groupsToAdd);
          const startDate = new Date(proj.start_date);
          const startValue = moment(startDate).valueOf();
          const endDate = !proj.end_date
            ? new Date(2050, 1, 1)
            : new Date(proj.end_date);
          const endValue = moment(endDate).valueOf();
          let color = "";
          switch (proj.project_state) {
            case "verde":
              color = "#7ede6d";
              break;
            case "rojo":
              color = "#E87272";
              break;
            case "amarillo":
              color = "#FAE269";
              break;
            case "upcomping":
              color = "#B0CFCB";
              break;
          }
          itemsToAdd.push({
            id: proj.id,
            group: proj.id,
            start: startValue,
            end: endValue,
            canMove: startValue > new Date().getTime(),
            canResize: "both",
            itemProps: {
              style: {
                borderRadius: 5,
                background: color,
              },
            },
            className:
              moment(startDate).day() === 6 || moment(startDate).day() === 0
                ? "item-weekend"
                : "",
          });
        });
        setItems(itemsToAdd);
      })
      .catch((error) => {
        setNotify({
          ...notify,
          isOpen: true,
          message: "No se pudieron cargar los datos de los proyectos",
          type: "error",
        });
        setFetchingError(true);
      });
  };
  useEffect(() => {
    fetchData();
  }, []);

  const handleItemClick = async (itemId) => {
    axiosInstance.get("/projects/" + itemId).then((response) => {
      var data = response.data.project;
      setProjectData(data);
      setOpenInfo(true);
    });
  };

  const defaultTimeStart = moment().startOf("day").toDate();
  const defaultTimeEnd = moment().startOf("day").add(30, "day").toDate();

  const onFilterChange = (e) => {
    setFilters((prevFilter) => ({
      ...prevFilter,
      [e.target.name]: e.target.value,
    }));
  };
  if (!isProjectView && !fetchingError) {
    return (
      <Fragment>
        <FilterForm
          onSubmit={(e) => {
            e.preventDefault();
            fetchData(filters);
          }}
          onClear={() => {
            setFilters({});
            fetchData();
          }}
          onInputChange={onFilterChange}
          project_state={filters.project_state}
          project_type={filters.project_type}
          organization={filters.organization}
        />
        {filteredData ? (
          <Timeline
            groups={groups}
            items={items}
            keys={keys}
            fullUpdate
            itemTouchSendsClick={true}
            dragSnap={60 * 60 * 24 * 1000} //dia
            itemHeightRatio={0.75}
            canMove={true} //se pueden mover
            canChangeGroup={false} //no se pueden "cambiar de renglon"
            canResize={"both"}
            defaultTimeStart={defaultTimeStart}
            defaultTimeEnd={defaultTimeEnd}
            timeSteps={customTimeSteps}
            onItemClick={handleItemClick}
            sidebarWidth={200}
          >
            <TimelineHeaders className="sticky">
              <SidebarHeader style={{}}>
                {({ getRootProps }) => {
                  return (
                    <div {...getRootProps()}>
                      <Switcher
                        onSwitch={onSwitch}
                        isProjectView={isProjectView}
                      />
                    </div>
                  );
                }}
              </SidebarHeader>
              <DateHeader unit="primaryHeader" />
              <DateHeader />
            </TimelineHeaders>
          </Timeline>
        ) : (
          <Notificacion notify={notify} setNotify={setNotify} />
        )}
        <Modal open={openInfo} onClose={handleInfoClose} disableEnforceFocus>
          <Box className={classes.modalInfo}>
            <IconButton
              aria-label="Close"
              onClick={handleInfoClose}
              className={classes.closeButton}
            >
              <CloseIcon />
            </IconButton>
            <InfoProyecto
              projectData={projectData}
              type={projectData.project_type}
              state={projectData.project_state}
            />
          </Box>
        </Modal>
      </Fragment>
    );
  }
  return <Notificacion notify={notify} setNotify={setNotify} />;
}
