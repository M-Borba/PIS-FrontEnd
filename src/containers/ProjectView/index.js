/**
 * Proyect View
 *
 * This is the container that has all actions and components related
 * to de Proyect View, were we manage our proyects.
 *
 */

import React, { Component } from "react";
import CustomTimeline from "./CustomTimeline";
import "react-calendar-timeline/lib/Timeline.css";

export default class ProjectView extends Component {
  render() {
    return (
      <div>
        <CustomTimeline />
      </div>
    );
  }
}
