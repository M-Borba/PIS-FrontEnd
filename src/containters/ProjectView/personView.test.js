import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ReactDOM from 'react-dom';
import ProjectView from "./index";
import React from "react";

describe("person timeline component tests", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ProjectView />, div);
  });
  it("renders correct data", () => {
    const { container } = render(
      <ProjectView
      />
    );
    // Matches the last snapshot taken
    expect(container).toMatchSnapshot();
  });
});
