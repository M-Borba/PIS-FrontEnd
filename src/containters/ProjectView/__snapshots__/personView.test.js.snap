// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`person timeline component tests renders correct data 1`] = `
<div>
  <div>
    <div
      class="react-calendar-timeline "
    >
      <div
        class="rct-header-root"
        style="display: flex; width: 100%;"
      >
        <div
          style="width: 150px;"
        />
        <div
          class="rct-calendar-header"
          style="overflow: hidden; width: calc(100% - 150px);"
        >
          <div
            style="height: 30px; position: relative; width: -450px;"
          >
            <div
              class="rct-dateHeader rct-dateHeader-primary"
              style="left: 39900px; width: -54750px; position: absolute;"
            >
              <span>
                21
              </span>
            </div>
          </div>
          <div
            style="height: 30px; position: relative; width: -450px;"
          >
            <div
              class="rct-dateHeader "
              style="left: 39900px; width: -54750px; position: absolute;"
            >
              <span>
                21
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="rct-outer"
        style="height: 900px;"
      >
        <div
          class="rct-sidebar"
          style="width: 150px; height: 900px;"
        >
          <div
            style="width: 150px;"
          >
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 1
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 2
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 3
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 4
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 5
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 6
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 7
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 8
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 9
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 10
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 11
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 12
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 13
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 14
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 15
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 16
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 17
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 18
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 19
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 20
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 21
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 22
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 23
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 24
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 25
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 26
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 27
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 28
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-even"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 29
            </div>
            <div
              class="rct-sidebar-row rct-sidebar-row-odd"
              style="height: 30px; line-height: 30px;"
            >
              Proyecto 30
            </div>
          </div>
        </div>
        <div
          class="rct-scroll"
          style="width: -150px; height: 920px; cursor: default; position: relative;"
        >
          <div
            style="position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px;"
          >
            <div
              class="rct-vertical-lines"
            >
              <div
                class="rct-vl"
                style="pointer-events: none; top: 0px; left: 39900px; width: -54750px; height: 900px;"
              />
            </div>
            <div
              class="rct-horizontal-lines"
            >
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-even "
                style="width: -450px; height: 30px;"
              />
              <div
                class="rct-hl-odd "
                style="width: -450px; height: 30px;"
              />
            </div>
            <div
              class="rct-items"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
